homeassistant/components/sms/notify.py

```diff
@@ -34,14 +34,29 @@ def __init__(self, gateway, number):
 
     def send_message(self, message="", **kwargs):
         """Send SMS message."""
-        # Prepare message data
-        # We tell that we want to use first SMSC number stored in phone
-        gammu_message = {
-            "Text": message,
-            "SMSC": {"Location": 1},
-            "Number": self.number,
+        smsinfo = {
+            "Class": -1,
+            "Unicode": False,
+            "Entries": [{"ID": "ConcatenatedTextLong", "Buffer": message}],
         }
         try:
-            self.gateway.SendSMS(gammu_message)
+            # Encode messages
+            encoded = gammu.EncodeSMS(smsinfo)  # pylint: disable=no-member
         except gammu.GSMError as exc:  # pylint: disable=no-member
-            _LOGGER.error("Sending to %s failed: %s", self.number, exc)
+            _LOGGER.error("Encoding message %s failed: %s", message, exc)
+            return
+
+        # Send messages
+        for encoded_message in encoded:
+            # Prepare message data
+            gammu_message = {
+                "Text": encoded_message["Text"],
+                # We tell that we want to use first SMSC number stored in phone
+                "SMSC": {"Location": 1},
+                "Number": self.number,
+            }
+            try:
+                # Actually send the message
+                self.gateway.SendSMS(gammu_message)
+            except gammu.GSMError as exc:  # pylint: disable=no-member
+                _LOGGER.error("Sending to %s failed: %s", self.number, exc)
```

Comment:
Should we continue sending on error? Don't we risk an incomplete message then?

Reply:
Yes, but it gets complex to manage, the benefit/cost ratio does not justify it.

Reply:
So then we should `return` here, right?
```suggestion
                _LOGGER.error("Sending to %s failed: %s", self.number, exc)
                return
```

Reply:
No, we want to keep trying.

Reply:
Ok.